### DISCUSSION TITLE
Scoping the export and re-import on another BookWyrm instance

### DISCUSSION THREAD

**Discussion by chdorner** [bookwyrm-social/bookwyrm#2684] [Ideas]
I have been thinking of taking on another reasonably large chunk of work which is the general topic of BookWyrm export/import and I wanted to run a few thoughts by you all.

I've collected existing Github issues together that are explicitly related to import/export from one BookWyrm instance to another:

- https://github.com/bookwyrm-social/bookwyrm/issues/2669
- https://github.com/bookwyrm-social/bookwyrm/issues/2666
- https://github.com/bookwyrm-social/bookwyrm/issues/2156
- https://github.com/bookwyrm-social/bookwyrm/issues/1012
- https://github.com/bookwyrm-social/bookwyrm/issues/2411

They can generally be categorised into two themes:

1. Improving/fixing the current export/import specifically related to books, their reading states, review, etc.
2. More general BookWyrm export/import to allow users to move instances, this can (should?) include additional objects like comments, quotes, and lists. Maybe followings, and probably much later followers.

I'm thinking of moving more towards that second one, but that would probably mean a different file structure, leaning towards JSON instead of CSV given the many more types of objects.

What do other people think? Is anybody staunchly on the side of only exporting bookshelves along with reviews in a CSV file?

On the positive side, this large piece of work lends itself fairly easily to smaller iterations. I could imagine starting with an initial version of exporting and importing the built-in shelves (to-read, reading, read, stopped reading) alongside reviews, and then adding additional objects like quotes and others one by one.

**Comment by mouse-reeve** [comment]
I think a JSON export would be great! I think the main advantage to CSV is that it's more widely portable, but I doubt other platforms are likely to support importing data from bookwyrm, and it would be better to be more able to transfer data between instances (a super high demand use case) than the use CSV specifically (which afaik no one is wedded to).

**Reply by chdorner** [reply]
That's a good point. Down the line we could offer two separate exports:
1. BookWyrm JSON format allowing maximum data portability
2. Taking one of the more common CSV formats for data portability between different service providers

I might start building the JSON format as an additional format and leave the existing CSV export as-is for now.

**Reply by mouse-reeve** [reply]
I actually think it would be better to sunset the CSV export if there's a workable JSON, since the CSV needs a lot of work to be viable, and I'd rather not get support requests for both. But also proceed, it sounds like you're on the right path!

**Comment by jmb** [comment]
Not 100% related to this (but definitely related to the generic import issues) I was interested to see if I could get my loan history from my Library BorrowBox account to load into Bookworm. I've made a start on some rough and ready code here... https://github.com/jmb/borrowbox  It would be great to have a generic import format that I could transform the data into to get it into BookWyrm. 😃

**Comment by jaschaurbach** [comment]
What I would love to have is more like an „automated“ move to another instance. 

I would love to register a new account, log into my old one, enter the new one in a field, verify to move to this account in the new one and then the instances talk to another moving the data.

Of course we still need all the exports and imports just because it is great to have them.

**Comment by sciencefictionspacecraft** [comment]
It’s bonkers that I can import my history from Goodreads easily into a new Bookwyrm account, but not my actual Bookwyrm history if I simply want to move accounts.
